src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -192,7 +192,18 @@ const std::vector<std::shared_ptr<AbstractExpression>>& AbstractLQPNode::column_
   return left_input()->column_expressions();
 }
 
-std::vector<std::shared_ptr<AbstractExpression>> AbstractLQPNode::node_expressions() const { return {}; }
+size_t AbstractLQPNode::node_expression_count() const { return 0u; }
+
+std::shared_ptr<AbstractExpression>& AbstractLQPNode::node_expression(const size_t idx) {
+  Fail("No node expressions in this node");
+}
+
+const std::shared_ptr<AbstractExpression>& AbstractLQPNode::node_expression(const size_t idx) const {
+  // Use const_cast<> here so that deriving classes don't need to implement a const and a non-const override of this
+  // function.
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
```

Comment:
Yes, convinced, we need to solve this better. The reason I used const_cast, as stated in the comment is: "Use const_cast<> here so that deriving classes don't need to implement a const and a non-const override of this function"

Reply:
This is solved via a AbstractLQPNode::node_expressions member now